vcd/resource_vcd_catalog_vapp_template.go

```diff
@@ -256,6 +294,35 @@ func resourceVcdCatalogVappTemplateDelete(_ context.Context, d *schema.ResourceD
 	return nil
 }
 
+func vappTemplateLeaseUpdate(vcdClient *VCDClient, vAppTemplate *govcd.VAppTemplate, d *schema.ResourceData) error {
+	var storageLease = vAppTemplate.VAppTemplate.LeaseSettingsSection.StorageLeaseInSeconds
+	rawLeaseSection1, ok := d.GetOk("lease")
+	if ok {
+		// We have a lease block
+		rawLeaseSection2 := rawLeaseSection1.([]interface{})
+		leaseSection := rawLeaseSection2[0].(map[string]interface{})
+		storageLease = leaseSection["storage_lease_in_sec"].(int)
+	} else {
+		// No lease block: we read the lease defaults from the Org
+		adminOrg, err := vcdClient.GetAdminOrgFromResource(d)
+		if err != nil {
+			return fmt.Errorf("error retrieving admin Org from parent Org in vApp Template %s: %s", vAppTemplate.VAppTemplate.Name, err)
+		}
+		if adminOrg.AdminOrg.OrgSettings == nil || adminOrg.AdminOrg.OrgSettings.OrgVAppTemplateSettings == nil {
+			return fmt.Errorf("error retrieving Org lease settings")
+		}
+		storageLease = *adminOrg.AdminOrg.OrgSettings.OrgVAppTemplateSettings.StorageLeaseSeconds
+	}
```

Comment:
I have a doubt about this `else` part. Before this PR introduced the `lease` block, we did nothing in terms of lease. Now, if `lease` is not present, we take Org's value and use it.
Does it mean that previous code was wrong due to lacking of this feature? Should this be done by VCD and it is not? Or it is just custom logic that we are introducing?

Reply:
Before, VCD applied the Org defaults without asking.
Now, we have an explicit lease, and we need to match it to the Org defaults to make the plan succeed.
The default would work just as well without it, but we need to take care of the hidden values in the state.

Reply:
> Before, VCD applied the Org defaults without asking.
Then we don't need the `else`, do we? VCD will apply the defaults, then they will be read back when Read comes:
```
leaseInfo, err := vAppTemplate.GetLease()
...
err = d.Set("lease", leaseData)
```
Actually I did a try by removing the `else` contents, and not setting any `lease` block, and the state was correctly read:

```
            "lease": [
              {
                "storage_lease_in_sec": 64800
              }
            ],
```
(I configured the Org to have 18h lease)

Reply:
OK. It seems that I overthought the operation. That `else` was not needed, and I removed it.
Thanks for checking it